packages/beacon-node/src/sync/backfill/backfill.ts

```diff
@@ -503,10 +503,10 @@ export class BackfillSync extends (EventEmitter as {new (): BackfillSyncEmitter}
    * the initialization point is the same block tree as the DB once backfill
    */
   private async checkIfCheckpointSyncedAndValidate(): Promise<void> {
-    if (this.syncAnchor.lastBackSyncedBlock == null) {
+    if (this.syncAnchor.lastBackSyncedBlock === null) {
       throw Error("Invalid lastBackSyncedBlock for checkpoint validation");
     }
-    if (this.wsCheckpointHeader == null) {
+    if (this.wsCheckpointHeader === null) {
```

Comment:
type of `this.wsCheckpointHeader` is `BackfillBlockHeader | null`. Same observation [here](https://github.com/ChainSafe/lodestar/pull/4647/files#r996684857) applies